# Switch light-mode RandomX code to executable permissions only after
# its instructions have been generated in writable memory.

Index: src/crypto/randomx/vm_compiled_light.cpp
--- src/crypto/randomx/vm_compiled_light.cpp.orig
+++ src/crypto/randomx/vm_compiled_light.cpp
@@ -56,6 +56,9 @@ namespace randomx {
 #		endif
 
 		compiler.generateProgramLight(program, config, datasetOffset);
+#		ifdef XMRIG_SECURE_JIT
+		compiler.enableExecution();
+#		endif
 
 		CompiledVm<softAes>::execute();
 	}
